### PR TITLE
feat: add support of custom fonts to card elements for Android

### PR DESCRIPTION
**Pull request by perqin**
## Summary
Add support of custom `fontFamily` to the CardField and CardForm elements for Android.

This PR uses `ReactTypefaceUtils.applyStyles` to load typeface from assets, which is the same as the approach in ReactTextView.

Note that the font of the dropdown options TextView in CardForm's country text input can't be changed by this PR, because those TextViews are created by [the adapter in the stripe-android library](https://github.com/stripe/stripe-android/blob/master/payments-core/src/main/java/com/stripe/android/view/CountryAdapter.kt).

## Motivation
#681 

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one: 
- [ ] I have added relevant documentation for my changes.
- [x] This PR does not result in any developer-facing changes.

### CODE REVIEW

**Comment by charliecruzan-stripe** [comment]
Hi @perqin ! Thanks for the contribution, sorry it's taking so long to get a review. I'll take a look this week!

**Comment by dqii** [comment]
@charliecruzan-stripe any chance you'd be able to have a look this week?

File: android/src/main/java/com/reactnativestripesdk/CardFormView.kt [hunk 1]
```diff
@@ -134,6 +135,12 @@ class CardFormView(context: ThemedReactContext) : FrameLayout(context) {
       cardFormViewBinding.cardMultilineWidget.expiryDateEditText,
       cardFormViewBinding.postalCode
     )
+    val inputLayoutBindings = setOf(
+      cardFormViewBinding.cardMultilineWidget.cardNumberTextInputLayout,
+      cardFormViewBinding.cardMultilineWidget.cvcInputLayout,
+      cardFormViewBinding.cardMultilineWidget.expiryTextInputLayout,
+      cardFormViewBinding.postalCodeContainer,
+    )
```
Comment (charliecruzan-stripe) [inline android/src/main/java/com/reactnativestripesdk/CardFormView.kt:143]: I think we can simplify the code in here just a little bit more:

let's use the same bindings for fontFamily that we do for placeholderColor. So here, let's change:
```suggestion
val placeholderTextBindings = setOf(
      multilineWidgetBinding.tlExpiry,
      multilineWidgetBinding.tlCardNumber,
      multilineWidgetBinding.tlCvc,
      cardFormViewBinding.postalCodeContainer,
    )
```

Then we can change the `placeholderColor?.let` block to:
```
placeholderColor?.let {
      for (binding in placeholderTextBindings) {
        binding.defaultHintTextColor = ColorStateList.valueOf(Color.parseColor(it))
      }
    }
```

and then I'll suggest the code changes below as well

File: android/src/main/java/com/reactnativestripesdk/CardFormView.kt [hunk 2]
```diff
@@ -159,9 +166,17 @@ class CardFormView(context: ThemedReactContext) : FrameLayout(context) {
       }
     }
     fontFamily?.let {
+      // Load custom font from assets, and fallback to default system font
+      val typeface = ReactTypefaceUtils.applyStyles(null, -1, -1, it.takeIf { it.isNotEmpty() }, context.assets)
       for (binding in editTextBindings) {
-        binding.typeface = Typeface.create(it, Typeface.NORMAL)
+        binding.typeface = typeface
+      }
+      for (binding in inputLayoutBindings) {
```
Comment (charliecruzan-stripe) [inline android/src/main/java/com/reactnativestripesdk/CardFormView.kt:174]:
```suggestion
      for (binding in placeholderTextBindings) {
```

see suggestion above for context

**Review by charliecruzan-stripe — CHANGES_REQUESTED** [review]
small changes, but then looks good to me! re request review once these changes are made please :)

**Comment by charliecruzan-stripe** [comment]
Also- you can remove the font file, that font file already exists at `example/src/assets/fonts/`

File: android/src/main/java/com/reactnativestripesdk/CardFormView.kt [hunk 3]
```diff
@@ -159,9 +166,17 @@ class CardFormView(context: ThemedReactContext) : FrameLayout(context) {
...
+        binding.typeface = typeface
       }
+      cardFormViewBinding.countryLayout.typeface = typeface
```
[the remark this answers is not in the document]
Comment (perqin) [reply android/src/main/java/com/reactnativestripesdk/CardFormView.kt:177]: This is needed to also set the font for the label and error text of the TextInputLayout. You could refer to the [documentation](https://developer.android.com/reference/com/google/android/material/textfield/TextInputLayout#setTypeface(android.graphics.Typeface)).

File: android/src/main/java/com/reactnativestripesdk/CardFormView.kt (hunk 2 above)
Comment (perqin) [reply android/src/main/java/com/reactnativestripesdk/CardFormView.kt]: Changed.

File: android/src/main/java/com/reactnativestripesdk/CardFormView.kt (hunk 1 above)
Comment (perqin) [reply android/src/main/java/com/reactnativestripesdk/CardFormView.kt]: Changed.

**Review by charliecruzan-stripe — APPROVED** [review]

**Pull request merged**